system/Commands/Database/ShowTableInfo.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+/**
+ * This file is part of CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace CodeIgniter\Commands\Database;
+
+use CodeIgniter\CLI\BaseCommand;
+use CodeIgniter\CLI\CLI;
+use Config\Database;
+
+/**
+ * Get table data if it exists in the database.
+ */
+class ShowTableInfo extends BaseCommand
+{
+    /**
+     * The group the command is lumped under
+     * when listing commands.
+     *
+     * @var string
+     */
+    protected $group = 'Database';
+
+    /**
+     * The Command's name
+     *
+     * @var string
+     */
+    protected $name = 'db:table';
+
+    /**
+     * the Command's short description
+     *
+     * @var string
+     */
+    protected $description = 'Retrieves the selected table information.';
+
+    /**
+     * the Command's usage
+     *
+     * @var string
+     */
+    protected $usage = 'db:table <table_name> [options]';
+
+    /**
+     * The Command's arguments
+     *
+     * @var array<string, string>
+     */
+    protected $arguments = [
+        'table_name' => 'The table name for show information.',
+    ];
+
+    /**
+     * The Command's options
+     *
+     * @var array<string, string>
+     */
+    protected $options = [
+        '--show' => 'Retrieves list the names of all database tables.',
+    ];
+
+    /**
+     * get table info
+     */
+    public function run(array $params)
+    {
+        // Connect to database
+        $db        = Database::connect();
+        $getTables = $db->listTables();
+        // The database does not have a table
+        if ($getTables === []) {
+            return CLI::error('Database has no tables!', 'light_gray', 'red');
+        }
+
+        // show all tables name
+        if (CLI::getOption('show')) {
+            CLI::write('list the names of all database tables : ', 'black', 'yellow');
+            CLI::write(implode(' , ', $getTables), 'black', 'blue');
+
+            return CLI::newLine();
+        }
+
+        $tableName = array_shift($params);
+        // table name correct.
+        if (in_array($tableName, $getTables, true)) {
+            CLI::write("Data of table {$tableName} : ", 'black', 'yellow');
+            $thead = $db->getFieldNames($tableName);
+            $tbody = $db->table($tableName)->get()->getResultArray();
```

Comment:
It gets all the records, but what if there are 1,000,000 records?

Reply:
In this regard, I will add two `$options`.
`--sort` for sort tabel rows. `ASC | DESC` I set it to **ASC** by default
ex . **use->** ` --sort DESC`
I do not use command `orderBy`. I use the commands for sorting arrays `(krsort ,usort ,...)`.

`--limitRows` limit the number of rows you would like returned . I set it to **10** by default. Which will be customizable.
ex. **use->** ` --limitRows 100`
This is the solution that comes to my mind.
Is it useful? Otherwise, please suggest a solution.

Reply:
Do you plan to add something another limit like `--limitColumns`?

Reply:
There is no camelCase option like `--limitRows` now.
It seems it is better to use all lowercase option.

Reply:
> Do you plan to add something another limit like `--limitColumns`?

The answer is **no** now. Also, in the future, the feedback may be such that it needs to be added. That's why I used `--limitRows`, not `--limit`.

> There is no camelCase option like `--limitRows` now. It seems it is better to use all lowercase option.

I realized, I use ``--limitrows``.

Reply:
I would like the option to be `--limit-rows`

Reply:
> I would like the option to be `--limit-rows`

@kenjis I agree with this. It seems more **readable**.
what is your opinion?

Reply:
If this command has a few options, shorter is better. E.g. just `--row` or `--rows`.

I also think `--limit-rows` is better than `--limitrows`.